src/main/java/uk/gov/hmcts/divorce/common/event/Applicant2SolicitorApplyForFinalOrder.java

```diff
@@ -0,0 +1,239 @@
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+
+        log.info("{} about to submit callback invoked for Case Id: {}", FINAL_ORDER_REQUESTED_APP2_SOL, details.getId());
+
+        CaseData data = details.getData();
+        data.getApplication().setPreviousState(beforeDetails.getState());
+
+        final List<String> errors = applyForFinalOrderService.validateApplyForFinalOrder(data, true);
+        if (!errors.isEmpty()) {
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .data(data)
+                .errors(errors)
+                .build();
+        }
+
+        CaseDetails<CaseData, State> updatedDetails = applyForFinalOrderService.applyForFinalOrderAsApplicant2(details);
+        CaseData updatedData = updatedDetails.getData();
+        var updatedFo = updatedData.getFinalOrder();
+
+        final var finalOrderFeeOrderSummary = updatedFo.getApplicant2SolFinalOrderFeeOrderSummary();
+
+        if (updatedFo.isSolicitorPaymentMethodPba()) {
+            final Optional<String> pbaNumber = Optional.ofNullable(updatedFo.getPbaNumbers())
+                .map(dynamicList -> dynamicList.getValue().getLabel());
+            if (pbaNumber.isPresent()) {
+                final PbaResponse response = paymentService.processPbaPayment(
+                    updatedData,
+                    updatedDetails.getId(),
+                    updatedData.getApplicant2().getSolicitor(),
+                    pbaNumber.get(),
+                    updatedFo.getApplicant2SolFinalOrderFeeOrderSummary(),
+                    updatedFo.getApplicant2SolFinalOrderFeeAccountReference()
+                );
+
+                if (response.getHttpStatus() == CREATED) {
+                    updatedFo.updateFinalOrderWithApp2SolPaymentDetails(
+                        finalOrderFeeOrderSummary,
+                        updatedData,
+                        response.getPaymentReference(),
+                        pbaNumber.get()
+                    );
+                } else {
+                    return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                        .data(details.getData())
+                        .errors(singletonList(response.getErrorMessage()))
+                        .build();
+                }
+            } else {
+                log.error(
+                    "PBA number not present when payment method is 'Solicitor fee account (PBA)' for CaseId: {}",
+                    details.getId());
+
+                return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                    .data(updatedData)
+                    .errors(singletonList("PBA number not present when payment method is 'Solicitor fee account (PBA)'"))
+                    .build();
+            }
+        }
+
+        updatedFo.setApplicant2SolAppliedForFinalOrder(YES);
```

Comment:
These could be set in the applyForFinalOrderService -> applyForFinalOrderAsApplicant2() method with an if statement in there? As it's called in the aboutToSubmit already

Reply:
Added new service method and case task